c2corg_ui/static/js/imageuploader.js

```diff
@@ -204,16 +192,27 @@ app.ImageUploaderController.prototype.upload_ = function() {
       });
 
       this.getImageMetadata_(file);
-      this.uploading[i] = this.api_.uploadImage(file);
-      this.promises.push(this.uploading[i]);
 
-      this.uploading[i].then(function(resp) {
-        console.log('100% uploaded! ' + file['metadata']['title'] + ' ' + i);
+      var canceller = this.q_.defer();
+      var promise = this.api_.uploadImage(file, canceller.promise, function(file, event) {
+        var progress = event.loaded / event.total;
+        file['progress'] = 100 * progress;
+      }.bind(this, file));
+      file['uploading'] = promise;
+      file['canceller'] = canceller;
+
+      promise.then(function(resp) {
+        file['metadata']['filename'] = resp['data']['filename'];
       }.bind(this), function(resp) {
-        this.alerts_.addError('error while uploading the image ' + resp);
-      }.bind(this), function(evt) { // handled by defer.notify()
-          //var progressPercentage = parseInt(100.0 * evt.loaded / evt.total, 10);
-      });
+        if (resp.status == -1) {
+          if (!file['manuallyAborted']) {
+            this.alerts_.addError('error while uploading the image : Timeout');
```

Comment:
The message probably has to be added to https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/templates/i18n.html#L132-L135 in order to be extracted and added to the PO files.